src/plugins/data_source_management/public/components/direct_query_data_sources_components/connection_detail/direct_query_connection_detail.tsx

```diff
@@ -70,7 +70,7 @@ export const DirectQueryDataConnectionDetail: React.FC<DirectQueryDataConnection
   setBreadcrumbs,
 }) => {
   const [observabilityDashboardsExists, setObservabilityDashboardsExists] = useState(false);
-  const checkIfSQLWorkbenchPluginIsInstalled = () => {
+  const checkIfObservabilityDashboardsPluginIsInstalled = () => {
```

Comment:
This function is used in multiple places, I'd suggest to extract it to a common place so to reuse it. Plus with @SuZhou-Joe's suggestion to use `http.fetch`
```typescript
  const checkIfObservabilityDashboardsPluginIsInstalled = () => {
    fetch('/api/status', {
      headers: {
        'Content-Type': 'application/json',
        'osd-xsrf': 'true',
        'accept-language': 'en-US,en;q=0.9,zh-CN;q=0.8,zh;q=0.7,zh-TW;q=0.6',
        pragma: 'no-cache',
        'sec-fetch-dest': 'empty',
        'sec-fetch-mode': 'cors',
        'sec-fetch-site': 'same-origin',
      },
      method: 'GET',
      referrerPolicy: 'strict-origin-when-cross-origin',
      mode: 'cors',
      credentials: 'include',
    })
      .then(function (response) {
        return response.json();
      })
      .then((data) => {
        for (let i = 0; i < data.status.statuses.length; ++i) {
          if (data.status.statuses[i].id.includes('plugin:observabilityDashboards')) {
            return true
          }
        }
        return false
      })
      .catch((error) => {
        notifications.toasts.addDanger(
          'Error checking Dashboards Observability Plugin Installation status.'
        );
        // eslint-disable-next-line no-console
        console.error(error);
      });
  };

```

Reply:
fixed.

Reply:
Just committed another version for using `http.get` plus a generic parms.